CMakeLists.txt

```diff
@@ -723,6 +723,13 @@ if(QMC_CUDA OR ENABLE_CUDA)
       set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} --allow-unsupported-compiler")
     endif()
     enable_language(CUDA)
+    if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 12.0)
+      if(CMAKE_CUDA_COMPILER_VERSION VERSION_EQUAL 11.7)
```

Comment:
I suggest the inequality on the CUDA version as well:

VERSION_LESS_EQUAL 11.7

and update the message
CUDA versions <=  11.7 do not support GCC versions >= 12. Please change...  e.g. use an older GCC version compatible with the CUDA version.